consul/src/main/java/com/linecorp/armeria/internal/consul/ConsulClient.java

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal.consul;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import javax.annotation.Nullable;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.MoreObjects;
+
+import com.linecorp.armeria.client.ClientDecoration;
+import com.linecorp.armeria.client.ClientOption;
+import com.linecorp.armeria.client.ClientOptions;
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.client.WebClientBuilder;
+import com.linecorp.armeria.client.retry.RetryRule;
+import com.linecorp.armeria.client.retry.RetryingClient;
+import com.linecorp.armeria.common.QueryParams;
+
+/**
+ * The Consul Client for accessing to consul agent API server.
+ */
+public final class ConsulClient {
+    /**
+     * Default Consul API SCHEME.
+     */
+    @VisibleForTesting
+    static final String DEFAULT_SCHEME = "http";
+
+    /**
+     * Default Consul HTTP API HOST.
+     */
+    @VisibleForTesting
+    static final String DEFAULT_HTTP_HOST = "localhost";
+
+    /**
+     * Default Consul HTTP API PORT.
+     */
+    private static final int DEFAULT_HTTP_PORT = 8500;
+
+    /**
+     * Default Consul API version.
+     */
+    private static final String DEFAULT_API_VERSION = "v1";
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    private static final ClientOptions retryingClientOptions =
+            ClientOptions.of(ClientOption.DECORATION.newValue(ClientDecoration.of(
+                    RetryingClient.newDecorator(RetryRule.failsafe(), 3))));
+
+    /**
+     * Returns a new {@link Builder}.
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    private volatile HealthClient healthClient;
```

Comment:
I think we don't need to lazily initialize `healthClient`. Because it is not used lazily. 
Please create this in the contructor.

Reply:
Yes.